### PR TITLE
[MM-44302] switch channel: add condition to remove unread styling for archived channels and filter archived channels from local data

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
switch channel: add the condition to remove unread styling for archived channels  and filter archived channels from local data

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-44302

#### Related Pull Requests
NA

#### Screenshots
|  before  |  after  |
|----|----|
| image-20220516-193656 | Screenshot 2023-01-09 at 6 10 30 PM |

#### Release Note
```release-note
* switch channel: add the condition to remove unread styling for archived channels  and filter archived channels from local data
```

### CODE REVIEW

**Review by esethna — APPROVED** [review]

File: components/suggestion/switch_channel_provider.jsx [hunk 1]
```diff
@@ -51,6 +51,8 @@ import {isGuest} from 'mattermost-redux/utils/user_utils';
 import {Preferences} from 'mattermost-redux/constants';
 import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
 
+import ProfilePicture from '../profile_picture';
```
Comment (michelengelen) [inline components/suggestion/switch_channel_provider.jsx:54]: can we import from `components/profile_picture` here please?

File: components/suggestion/switch_channel_provider.jsx [hunk 2]
```diff
@@ -252,7 +254,7 @@ class SwitchChannelSuggestion extends Suggestion {
                 {icon}
                 <div className='suggestion-list__ellipsis suggestion-list__flex'>
                     <span className='suggestion-list__main'>
-                        <span className={item.unread ? 'suggestion-list__unread' : ''}>{name}</span>
+                        <span className={item.unread && !channelIsArchived ? 'suggestion-list__unread' : ''}>{name}</span>
```
Comment (michelengelen) [inline components/suggestion/switch_channel_provider.jsx:257]: could you please utilize the `classnames` package here?

**Review by michelengelen — APPROVED** [review]
please adjust the 2 small comments I had on this ... LGTM otherwise.

File: components/suggestion/switch_channel_provider.jsx [hunk 3]
```diff
@@ -54,6 +53,8 @@ import {isGuest} from 'mattermost-redux/utils/user_utils';
 import {Preferences} from 'mattermost-redux/constants';
 import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
 
+import classnames from 'classnames';
```
Comment (AshishDhama) [inline components/suggestion/switch_channel_provider.jsx:56]: although this an duplicate but some how breaking the eslint

File: components/suggestion/switch_channel_provider.jsx [hunk 4]
```diff
@@ -7,7 +7,6 @@ import {connect} from 'react-redux';
 import classNames from 'classnames';
```
Comment (AshishDhama) [inline components/suggestion/switch_channel_provider.jsx:7]: @hmhealey `classNames` import is there, but still ESlint is failing with this error

Comment (hmhealey) [reply components/suggestion/switch_channel_provider.jsx:7]: You've actually got a casing issue. The import is `classNames` but you're using `classnames` in your new code

**Review by michelengelen — APPROVED** [review]

File: components/suggestion/switch_channel_provider.jsx (hunk 4 above)
Comment (AshishDhama) [reply components/suggestion/switch_channel_provider.jsx:7]: 🤦‍♂️, thanks man

**Comment by Willyfrog** [comment]
@AshishDhama @jgilliam17 what are the NS to merge this?

**Review by jgilliam17 — APPROVED** [review]
Thanks @AshishDhama 
Tested, looks good to merge.
Verified archived channels no longer have unread styling on channel switcher.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/294235-956fc28-onprem-ent-webapp-pr-11930-mm-ee-test:956fc28/mochawesome.html) - no PR related failures.

**Comment by jgilliam17** [comment]
@Willyfrog or @hmhealey for second dev review - thanks 🙂

**Comment by AshishDhama** [comment]
@Willyfrog waiting for your review on this one.

**Review by Willyfrog — APPROVED** [review]

**Pull request merged**